include/rocksdb/utilities/object_registry.h

```diff
@@ -62,10 +79,48 @@ class ObjectLibrary {
     FactoryFunc<T> factory_;
   };  // End class FactoryEntry
  public:
+  // Loads the ObjectLibrary specified by the input value into result
+  // Libraries can be "default", "local", or "dynamic".
+  // - "Default" says to use the ObjectLibrary::Default()
+  // - "local" uses methods from the current executable.  Local libraries have
+  //        libraries have the following control parameters:
+  //      - "method"  specifies the method in the executable to invoke
+  //      - "arg"     specifies the argument to pass to that method
+  // - "dynamic" uses methods from a dynamically loaded library.  Dynamic
+  //        libraries have the following control parameters:
+  //      - "library" specifies the library to load
+  //      - "method"  specifies the method in that library to invoke
+  //      - "arg"     specifies the argument to pass to that method
+  // @param config_options Controls how the library is loaded
+  // @param value The name and optional properties describing the library
+  //      to load.
+  // @param result On success, returns the loaded library
+  // @return OK if the library  was successfully loaded.
+  // @return not-OK if the load failed.
+  static Status CreateFromString(const ConfigOptions& config_options,
+                                 const std::string& value,
+                                 std::shared_ptr<ObjectLibrary>* result);
+  static const char* Type() { return "ObjectLibrary"; }
+  virtual ~ObjectLibrary() {}
+
   // Finds the entry matching the input name and type
   const Entry* FindEntry(const std::string& type,
                          const std::string& name) const;
-  void Dump(Logger* logger) const;
```

Comment:
Dump is in the public API, if some one is using it, how they can write out the info in registry to the log?